src/management-system-v2/lib/useColumnWidth.tsx

```diff
@@ -1,206 +1,255 @@
-    // event.stopPropagation();
-    event.preventDefault();
-    resizingColumn.current = key;
-    // @ts-ignore
-    columnRef.current = event.target.closest('.ColumnWidthParent') as HTMLElement;
-    startX.current = event.clientX;
-
-    /* Create and append the indicator div */
-    if (document) {
-      const tableRef = columnRef.current?.closest('table');
-
-      indicatorRef.current = document.createElement('div');
-      indicatorRef.current.style.position = 'absolute';
-      indicatorRef.current.style.height = `${tableRef!.offsetHeight}px`;
-      indicatorRef.current.style.width = '1px';
-      indicatorRef.current.style.backgroundColor = 'grey';
-      indicatorRef.current.style.left = `${event.clientX}px`;
-      document.body.appendChild(indicatorRef.current);
-      indicatorRef.current.style.top = `${tableRef!.getBoundingClientRect().top}px`;
-      indicatorRef.current.style.zIndex = '1000';
-    }
-  }, []);
+    if (onlySelectedColumnsCauseRerender.current && columns.length === columnsInPreferences.length)
+      return;
 
-  const handleMouseMove = useCallback((event: MouseEvent) => {
-    if (indicatorRef.current) {
-      indicatorRef.current.style.left = `${event.clientX}px`;
-    }
-  }, []);
+    let newColumns = columns.map((column: any) => {
+      const columnInPreferences = columnsInPreferences.find(
+        (col: any) => col.name === column.title,
+      );
+      if (!columnInPreferences) return column;
+
+      return {
+        ...column,
+        width: columnInPreferences.width,
+      };
+    });
 
-  const handleMouseUp = useCallback(
-    (event: MouseEvent) => {
-      if (resizingColumn.current) {
-        /* Calculate new Width */
-        const currentWidth = columnRef.current?.offsetWidth || 0;
-        const newWidth = currentWidth + event.clientX - startX.current;
+    setResizeableColumns(newColumns);
+
+    if (!onlySelectedColumnsCauseRerender.current && document) {
+      setTimeout(() => {
+        /* Hydrating */
+        /* Replace 'auto' width with actual px values */
+        const resizeableElements = document.querySelectorAll('.react-resizable');
+        const widthsOfResizeableElements = Array.from(resizeableElements).map(
+          (element: any) => element.getBoundingClientRect().width,
+        );
+        /* Get the widths */
+        const widths = columns.map((column: any, index: number) => {
+          /* Check if not resizeable */
+          if (notResizeabel.includes(column.key)) return column.width || minWidth;
+
+          return widthsOfResizeableElements.shift() || minWidth;
+        });
+
+        /* Replace width */
+        newColumns = newColumns.map((column: any, index: number) => {
+          return {
+            ...column,
+            width: widths[index],
+          };
+        });
 
-        /* Update UI */
-        setColumnWidths((old) => ({ ...old, [resizingColumn.current]: newWidth }));
+        setResizeableColumns(newColumns);
+      }, 1_000);
```

Comment:
This is a workaround to keep antdesigns 'auto' width. It replaces the auto values with the actual pixel values after everything was rendered.